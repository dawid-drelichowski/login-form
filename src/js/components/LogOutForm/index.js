// @flow
import React, {Component} from 'react'
import type {Node} from 'react'

export default class LogOutForm extends Component<{onLogOut?: () => void}> {
  onSubmit = (event: SyntheticEvent<HTMLInputElement>): void => {
    const onLogOut: () => void = this.props.onLogOut ? this.props.onLogOut : () => {}
    event.preventDefault()
    onLogOut()
  }
  render (): Node {
    return <form method="post" onSubmit={this.onSubmit}>
      <fieldset>
        <div>You are logged in!</div>
        <input type="submit" value="Logout"/>
      </fieldset>
    </form>
  }
}
