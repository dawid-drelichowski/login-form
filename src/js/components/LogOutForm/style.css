.log-out-form {
  background: #fff;
  max-width: 300px;
  padding: 30px;
  margin: 10% auto 0;
  box-shadow: 0 0 20px 0 rgba(0, 0, 0, 0.2), 0 5px 5px 0 rgba(0, 0, 0, 0.24);
}

.log-out-form .field-set {
  border: 0;
  text-align: center;
}

.log-out-form .field-set .text {
  font-size: 1.5rem;
  margin: 15px;
}

.log-out-form .field-set .button {
  text-transform: uppercase;
  background: #4caf50;
  width: 100%;
  border: 0;
  padding: 10px;
  color: #fff;
  font-size: 1rem;
  cursor: pointer;
}
