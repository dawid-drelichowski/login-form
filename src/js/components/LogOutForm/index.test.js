import React from 'react'
import LogOutForm from './index'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

describe('Log out form component', () => {
  function getRenderedComponent () {
    return renderer.create(<LogOutForm />)
  }

  function getShallowComponent (onLogOut = () => {}) {
    return shallow(<LogOutForm onLogOut={onLogOut}/>)
  }

  it('should render', () => {
    expect(getRenderedComponent().toJSON()).toMatchSnapshot()
  })

  it('should call "onLogOut" callback when the form submitted', () => {
    const onLogOut = jest.fn(),
      form = getShallowComponent(onLogOut)

    form.simulate('submit', {preventDefault: jest.fn()})
    expect(onLogOut).toHaveBeenCalledTimes(1)
  })
})
