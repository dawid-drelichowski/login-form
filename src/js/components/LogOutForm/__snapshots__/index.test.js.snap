// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log out form component should render 1`] = `
<form
  method="post"
  onSubmit={[Function]}
>
  <fieldset>
    <div>
      You are logged in!
    </div>
    <input
      type="submit"
      value="Logout"
    />
  </fieldset>
</form>
`;
